components/n-ui/ads/js/ads-metrics.js

```diff
@@ -0,0 +1,80 @@
+import nUIFoundations from 'n-ui-foundations';
+import { utils as oAdsUtils } from 'o-ads';
+
+// Probability that a page view is chosen to be send ads-related metrics
+const METRICS_SAMPLE_SIZE = 0.1;
+
+let userSendsMetrics;
+
+const inAdsMetricsSample = function () {
+	// We cache the value in order to be consistent with the user
+	// allocation throughout a page view
+	if (typeof userSendsMetrics !== 'undefined') {
+		return userSendsMetrics;
+	}
+
+	userSendsMetrics = (Math.random() < METRICS_SAMPLE_SIZE);
+	return userSendsMetrics;
+};
+
+const metricsDefinitions = [
+	{
+		spoorAction: 'page-initialised',
+		triggers: ['serverScriptLoaded'],
+		marks: [
+			'initialising',
+			'IVTComplete',
+			'adsAPIComplete',
+			'initialised',
+			'serverScriptLoaded',
+		]
+	},
+	{
+		spoorAction: 'krux',
+		triggers: ['kruxKuidAck', 'kruxKuidError', 'kruxConsentOptinFailed'],
+		marks: [
+			'kruxScriptLoaded',
+			'kruxConsentOptinOK',
+			'kruxConsentOptinFailed',
+			'kruxKuidAck',
+			'kruxKuidError',
+		]
+	},
+	{
+		spoorAction: 'slot-requested',
+		triggers: ['slotGoRender'],
+		marks: [
+			'slotReady',
+			'slotCanRender',
+			'slotGoRender',
+		],
+		multiple: true
+	},
+	{
+		spoorAction: 'slot-rendered',
+		triggers: ['slotRenderEnded'],
+		marks: [
+			'slotRenderStart',
+			'slotExpand',
+			'slotRenderEnded',
+		],
+		multiple: true
+	}
+];
+
+function sendMetrics (eventPayload) {
+	if (inAdsMetricsSample()) {
+		nUIFoundations.broadcast('oTracking.event', eventPayload);
+	}
+}
+
+function setupAdsMetrics () {
+	oAdsUtils.setupMetrics(metricsDefinitions, sendMetrics);
```

Comment:
To me it's a little confusing passing `sendMetrics` as a callback. In `o-ads.utils` there is also a function called sendMetrics, which calls this `callback` on the last line. A neater solution could be passing the `inAdsMetricsSample` as a parameter here, delete the `sendMetrics` function above and use the `sendMetrics` function in o-ads to do the broadcasting of the event.

Reply:
The main reason for having `sendMetrics` as a callback is because we don't want users of `o-ads` to be force to use Spoor API for their metrics. This approach enables greater flexibility. 

This not only keeps `o-ads` unaware of `Spoor API` but it also prevents it from needing `n-ui-foundations` as a dependency.

Adding the metrics sampling to `o-ads` is already on my radar. I plan to add it as an optional config parameter on the metrics config objects. I just didn't want to overdo this PR, specially since we want to start collecting metrics asap.

Reply:
Ah ok cool makes sense. In that case, could we rename the `sendMetrics` and `callback` variables in the o-ads.utils? I'll message you about this separately as it's not relevant to this PR. Good work!